src/mse.ts

```diff
@@ -188,43 +179,49 @@ export class MSERep extends EventEmitter implements MSE {
 
 	// Rundown basics task
 	async createRundown(profileName: string, playlistID?: string, description?: string): Promise<VRundown> {
-		let playlistExists = false
-		const date = new Date()
-		description = description ? description : `Sofie Rundown ${date.toISOString()}`
+		await this.assertProfileExistsForName(profileName)
+
+		description = description ? description : `Sofie Rundown ${new Date().toISOString()}`
+		playlistID = playlistID ? playlistID.toUpperCase() : uuid.v4().toUpperCase()
+
+		if (!(await this.doPlaylistExistForPlaylistId(playlistID, profileName))) {
+			await this.createNewPlaylist(playlistID, description, profileName)
+		}
+		return new Rundown(this, profileName, playlistID, description)
+	}
+
+	private async assertProfileExistsForName(profileName: string): Promise<void> {
 		try {
 			await this.pep.get(`/config/profiles/${profileName}`, 1)
 		} catch (err) {
 			throw new Error(
 				`The profile with name '${profileName}' for a new rundown does not exist. Error is: ${getPepErrorMessage(err)}.`
 			)
 		}
-		if (playlistID) {
-			try {
-				const playlist = await this.getPlaylist(playlistID.toUpperCase())
-				if (!playlist.profile.endsWith(`/${profileName}`)) {
-					throw new Error(
-						`Referenced playlist exists but references profile '${playlist.profile}' rather than the given '${profileName}'.`
-					)
-				}
-				playlistExists = true
-			} catch (err) {
-				if (getPepErrorMessage(err).startsWith('Referenced playlist exists but')) {
-					throw err
-				}
-				playlistExists = false
+	}
+
+	private async doPlaylistExistForPlaylistId(playlistID: string, profileName: string): Promise<boolean> {
```

Comment:
Same as above, but perhaps _does_ is more gramatically correct than _do_, unless there are some rules to use the latter?
```suggestion
	private async doesPlaylistExist(playlistID: string, profileName: string): Promise<boolean> {
```

Reply:
Updated